### PR TITLE
A3C Blog Post

### PR DESCRIPTION
**Pull request by raymond-yuan**
Created PR for a3c for cartpole blog post.

### CODE REVIEW

File: research/a3c_blogpost/a3c_cartpole.py [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+import argparse
+import matplotlib.pyplot as plt
+
+
+import tensorflow as tf
+from tensorflow.python import keras
+from tensorflow.python.keras import layers
+
+tf.enable_eager_execution()
+print("Eager execution: {}".format(tf.executing_eagerly()))
```
Comment (alextp) [inline research/a3c_blogpost/a3c_cartpole.py]: Redundant print?

File: research/a3c_blogpost/a3c_cartpole.py [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+def str2bool(v):
```
Comment (alextp) [inline research/a3c_blogpost/a3c_cartpole.py]: I think this is nice on production code but not so nice on code that we expect people to read carefully. Just use type=bool

File: research/a3c_blogpost/a3c_cartpole.py [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+  def call(self, inputs):
+    # Forward pass
+    x = self.dense1(inputs)
+    logits = self.policy_logits(x)
+    v1 = self.dense2(inputs)
+    values = self.values(v1)
+    return logits, values
+
+
+def tf_wrap(np_array, dtype=np.float32):
```
Comment (alextp) [inline research/a3c_blogpost/a3c_cartpole.py]: Do you need this? I think a tf.cast(np_array, tf.dtype) should do the trick if a tf.convert_to_tensor(np_array, tf_dtype) doesn't work

File: research/a3c_blogpost/a3c_cartpole.py [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+
+  def train(self):
+    if args.algorithm == 'random':
+      random_agent = RandomAgent(self.game_name, args.max_eps)
+      random_agent.run()
+      return
+
+    res_queue = Queue()
+
+    # We run the algorithm on cpu only!
```
Comment (alextp) [inline research/a3c_blogpost/a3c_cartpole.py]: Why?

File: research/a3c_blogpost/a3c_cartpole.py (hunk 2 above)
Comment (MarkDaoust) [reply research/a3c_blogpost/a3c_cartpole.py]: One option is:

```
parser.add_argument("--no_train", action="store_false", dest="train", default=True)
```

The only advantage is that it's a tiny bit less code.

File: research/a3c_blogpost/a3c_cartpole.py [hunk 5]
```diff
@@ -0,0 +1,365 @@
+import os
+os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"   # see issue #152
```
Comment (alextp) [inline research/a3c_blogpost/a3c_cartpole.py]: Does the device order matter if we're setting visible devices to empty?

Comment (raymond-yuan) [reply research/a3c_blogpost/a3c_cartpole.py]: I was referring to https://github.com/keras-team/keras/issues/4613, but that line doesn't seem to affect anything. Removed.

**Comment by alextp** [comment]
Will let Mark or Francois merge when they approve
